pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -840,6 +844,20 @@ public LLRealtimeSegmentDataManager(RealtimeSegmentZKMetadata segmentZKMetadata,
     // Create field extractor
     _fieldExtractor = FieldExtractorFactory.getPlainFieldExtractor(schema);
     makeConsumerWrapper();
+
+    // Partitioners
+    _column2PartitionersMap = indexingConfig.getPartitioners();
+    if (_column2PartitionersMap != null && !_column2PartitionersMap.isEmpty()) {
+      try {
+        _nPartitions = _consumerWrapper.getPartitionCount(_kafkaTopic, /*maxWaitTimeMs=*/5000L);
+        _segmentPartitionConfig = createPartitionConfig();
+        _realtimeSegment.setSegmentPartitionConfig(_segmentPartitionConfig);
+      } catch (Exception e) {
+        segmentLogger.warn("Couldn't get number of partitions in 5s, not using partition config");
```

Comment:
Please call makeConsumerWrapper() again when the exception is fielded. The consumer wrapper is a stateful object, and we have found it better to create a new one when experiencing transient Kafka failures. In this case, whether the exception is transient or permanent, it is better to create a new consumer wrapper. See other uses of this method and the conditions in which they are called.